### PR TITLE
fix: safari compatibility

### PR DESCRIPTION
**Pull request by fundon**

### CODE REVIEW

**Comment by fundon** [comment]
* **#7130** <a href="https://app.graphite.dev/github/pr/toeverything/blocksuite/7130?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @fundon and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: packages/framework/inline/src/services/range.ts [hunk 1]
```diff
@@ -73,9 +74,10 @@ export class RangeService<TextAttributes extends BaseTextAttributes> {
       // range change may happen before the editor is prepared
       await this.editor.waitForUpdate();
       // improve performance
-      requestIdleCallback(() => {
-        this.editor.requestUpdate(false);
-      });
+      REQUEST_IDLE_CALLBACK_ENABLED &&
```
Comment (Saul-Mirone) [inline packages/framework/inline/src/services/range.ts:77]: If requestIdleCallback is not supported, maybe we should call this with setTimeout or Promise.resolve. @Flrande what do you think?

Comment (Flrande) [reply packages/framework/inline/src/services/range.ts:77]: `Promise.resolve` look good to me

Comment (fundon) [reply packages/framework/inline/src/services/range.ts:77]: Change to:

```ts
      if (REQUEST_IDLE_CALLBACK_ENABLED) {
        requestIdleCallback(() => {
          this.editor.requestUpdate(false);
        });
      } else {
        await Promise.resolve(this.editor.requestUpdate(false));
      }
```

**Pull request merged**